File: project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/model/link/Course.java
package org.babyfish.jimmer.sql.model.link;

import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.Id;
import org.babyfish.jimmer.sql.ManyToManyView;
import org.babyfish.jimmer.sql.OneToMany;

import java.util.List;

@Entity
public interface Course {

    @Id
    long id();

    String name();

    int academicCredit();

    @OneToMany(mappedBy = "course")
    List<LearningLink> learningLinks();

    @ManyToManyView(prop = "learningLinks", deeperProp = "student")
    // or @ManyToManyView(prop = "learningLinks")
    List<Student> students();

    @OneToMany(mappedBy = "nextCourse")
    List<CourseDependency> prevCourseDependencies();

    @ManyToManyView(prop = "prevCourseDependencies", deeperProp = "prevCourse")
    List<Course> prevCourses();

    @OneToMany(mappedBy = "prevCourse")
    List<CourseDependency> nextDependencies();

    @ManyToManyView(prop = "nextDependencies", deeperProp = "nextCourse")
    List<Course> nextCourses();
}

File: project/jimmer-sql-kotlin/src/test/kotlin/org/babyfish/jimmer/sql/kt/model/link/Course.kt
package org.babyfish.jimmer.sql.kt.model.link

import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.Id
import org.babyfish.jimmer.sql.ManyToManyView
import org.babyfish.jimmer.sql.OneToMany

@Entity
interface Course {

    @Id
    val id: Long

    val name: String

    val academicCredit: Int

    @OneToMany(mappedBy = "course")
    val learningLinks: List<LearningLink>

    @ManyToManyView(prop = "learningLinks", deeperProp = "student")
    val students: List<Student>
}